docs/changelog.md

```diff
@@ -154,7 +154,36 @@ To disable the `drop()` step optimization and maintain the previous behavior, us
 query.batch.drop-step-mode=none
 ```
 
-### Version 1.0.1 (Release Date: ???)
+##### Lazy Loading for Relations
+
+The new transaction configuration option is added `lazyLoadRelations()` which sets lazy-load for all properties and edges 
+of the vertex. If enabled, then ids and values are deserialized upon demand.  
+When enabled, it can lead to a performance improvement on large-scale read operations, if only certain types of relations are being read from the vertex.  
+See performance comparison in [GitHub PR #4343](https://github.com/JanusGraph/janusgraph/pull/4343/#issuecomment-2079839853).
+
+##### Text predicates support extended for remote connections
+
+The following text predicates can now be used with remote connections:
+
+* textNotContains
+* textNotContainsFuzzy
+* textNotContainsPrefix
+* textNotContainsRegex
+* textContainsPhrase
+* textNotContainsPhrase
+* textNotFuzzy
+* textNotPrefix
+* textNotRegex
+
+##### Vertex mutation optimizations
+
+The following improvements are made for adding new vertex or updating vertex properties:
+
+* Improved operations to detect vertex changes during transaction-commit from O(N) to O(1);
+* Optimizing properties search by key for newly-created vertex;
+* Optimizing previous property/edge search during vertex update;
```

Comment:
**❓ question:** ‏Is there one or more PRs associated with those optimizations we could reference? If not in the doc itself, maybe here as a PR comment?

Reply:
The PR is: https://github.com/JanusGraph/janusgraph/pull/4292
I just added it as a second commit to not trigger full CI

Reply:
Seems CI is triggered anyways. I guess it's per PR not per commit. Anyways, I will squash all commits when merging them into the `master` branch